distribution/pull_v2_unix.go

```diff
@@ -5,9 +5,14 @@ package distribution
 import (
 	"github.com/docker/distribution"
 	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/docker/distribution/xfer"
 )
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
+	if ld.src.MediaType == schema2.MediaTypeForeignLayer {
```

Comment:
This seems like an stopgap to me and I don't really like it. I would rather leave it the way it is with the client timing out eventually (and eventually Linux will have support for foreign layers soon...).